announcement/announcement-impl/impl/src/java/org/sakaiproject/announcement/impl/SiteEmailNotificationAnnc.java

```diff
@@ -402,16 +419,43 @@ protected String getFromAddress(Event event)
 					if ((userEmail != null) && (userEmail.trim().length()) == 0) userEmail = null;
 					
 				} catch (UserNotDefinedException e) {
-					log.warn("Failed to load user from announcement header: {}. Will send from no-reply@{} instead.", userId, ServerConfigurationService.getServerName());
+					log.warn("Failed to load user from announcement header: {}. Will send with no-reply@{} instead.", userId, ServerConfigurationService.getServerName());
 				}
 				
 				// some fallback positions
 				if (userEmail == null) userEmail = ServerConfigurationService.getString("setup.request","no-reply@" + ServerConfigurationService.getServerName());
 				if (userDisplay == null) userDisplay = ServerConfigurationService.getString("ui.service", "Sakai");
-				from="From: \"" + userDisplay + "\" <" + userEmail + ">";
+
+				address = field == AddressField.FROM ? "From: \"" : "To: \"";
+				// 'From' should display the user; 'To' should display the site title; if the title is unavailable, fallback to the user
+				String display = (field == AddressField.FROM || StringUtils.isBlank(title)) ? userDisplay : title;
+				address = address + display + "\" <" + userEmail + ">";
 		}
 		
-		return from;
+		return address;
+	}
+
+	/**
+	 * Format the announcement notification from address.
+	 *
+	 * @param event
+	 *        The event that matched criteria to cause the notification.
+	 * @return the announcement notification from address.
+	 */
+	protected String getFromAddress(Event event)
```

Comment:
This probably could be private any reason to not just remove this method and use getAddress()?

Reply:
Agreed; amendment incoming.